packages/app/obojobo-document-engine/src/scripts/oboeditor/components/toolbars/file-toolbar-viewer.js

```diff
@@ -46,14 +58,38 @@ const FileToolbarViewer = props => {
 			)
 		})()
 
-		const insertMenuItems = insertableItems.map(item => ({
-			name: item.name,
-			action: () => {
-				Transforms.insertNodes(editor, item.cloneBlankNode())
-				ReactEditor.focus(editor)
-			},
-			disabled: isInsertDisabledForItem(selectedNodes, item.name, sel)
-		}))
+		const insertMenuItems = insertableItems.map(item => {
+			return {
+				name: item.name,
+				action: () => {
+					const path = Editor.path(editor, editor.selection, { edge: 'start' })
+					const prevChildrenCount = editor.children.length
+
+					Transforms.insertNodes(editor, item.cloneBlankNode())
+
+					// Since inserting a node inside a figure caption can sometimes
+					// cause figure to be duplicated after the caption is split, we
+					// need to convert the duplicated figure below to a text node.
+					if (
+						containsFigureNode(selectedNodes) &&
+						editor.children.length !== prevChildrenCount + 1
+					) {
+						Transforms.setNodes(
+							editor,
+							{
+								type: TEXT_NODE,
+								content: {}
+							},
+							{
+								at: [path[0] + 2]
```

Comment:
This breaks when splitting a figure inside a QuestionBank Question (for example). I think that's because while the path might be `[4]` on a page, inside a QB Question it might be `[4, 1, 2]`. In that example you'd need to transform the node at `[4, 1, 4]`

Reply:
This code is also very specific, need to make sure we make an issue to modularize this